<a href="https://colab.research.google.com/github/ArchieRoche/Iphone-India-Sales-Insights-on-Flipkart/blob/main/Ecommerce_and_Apple_iphone_sales_data_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

What are the top 10 highest rated iphone on the ecomm site Flipkart?
how many ratings do the highest rated iphones on Flipkart have?
Which iphone has the highest bumber of reviews on FlipKart?
What is the relationship between the sales price of iphones and number of ratings on FlipKart?
What is the relationship between discount percentage and the number of ratings of iphones on FlipKart?
Can you figure out the least expensive and the most expensive iPhones in the indian market alongwith all their specifications?


In [6]:
import pandas as pd
import numpy as np
import plotly.express as px
import plotly.graph_objects as go

In [7]:
from google.colab import files
data_to_load = files.upload()


Saving apple_products.csv to apple_products.csv


In [8]:
data = pd.read_csv('apple_products.csv')

In [9]:
data

,Product Name,Product URL,Brand,Sale Price,Mrp,Discount Percentage,Number Of Ratings,Number Of Reviews,Upc,Star Rating,Ram
0,"APPLE iPhone 8 Plus (Gold, 64 GB)",https://www.flipkart.com/apple-iphone-8-plus-g...,Apple,49900,49900,0,3431,356,MOBEXRGV7EHHTGUH,4.6,2 GB
1,"APPLE iPhone 8 Plus (Space Grey, 256 GB)",https://www.flipkart.com/apple-iphone-8-plus-s...,Apple,84900,84900,0,3431,356,MOBEXRGVAC6TJT4F,4.6,2 GB
2,"APPLE iPhone 8 Plus (Silver, 256 GB)",https://www.flipkart.com/apple-iphone-8-plus-s...,Apple,84900,84900,0,3431,356,MOBEXRGVGETABXWZ,4.6,2 GB
3,"APPLE iPhone 8 (Silver, 256 GB)",https://www.flipkart.com/apple-iphone-8-silver...,Apple,77000,77000,0,11202,794,MOBEXRGVMZWUHCBA,4.5,2 GB
4,"APPLE iPhone 8 (Gold, 256 GB)",https://www.flipkart.com/apple-iphone-8-gold-2...,Apple,77000,77000,0,11202,794,MOBEXRGVPK7PFEJZ,4.5,2 GB
...,...,...,...,...,...,...,...,...,...,...,...
57,"APPLE iPhone SE (Black, 64 GB)",https://www.flipkart.com/apple-iphone-se-black...,Apple,29999,39900,24,95909,8161,MOBFWQ6BR3MK7AUG,4.5,4 GB
58,"APPLE iPhone 11 (Purple, 64 GB)",https://www.flipkart.com/apple-iphone-11-purpl...,Apple,46999,54900,14,43470,3331,MOBFWQ6BTFFJKGKE,4.6,4 GB
59,"APPLE iPhone 11 (White, 64 GB)",https://www.flipkart.com/apple-iphone-11-white...,Apple,46999,54900,14,43470,3331,MOBFWQ6BVWVEH3XE,4.6,4 GB
60,"APPLE iPhone 11 (Black, 64 GB)",https://www.flipkart.com/apple-iphone-11-black...,Apple,46999,54900,14,43470,3331,MOBFWQ6BXGJCEYNY,4.6,4 GB


In [10]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62 entries, 0 to 61
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Product Name         62 non-null     object 
 1   Product URL          62 non-null     object 
 2   Brand                62 non-null     object 
 3   Sale Price           62 non-null     int64  
 4   Mrp                  62 non-null     int64  
 5   Discount Percentage  62 non-null     int64  
 6   Number Of Ratings    62 non-null     int64  
 7   Number Of Reviews    62 non-null     int64  
 8   Upc                  62 non-null     object 
 9   Star Rating          62 non-null     float64
 10  Ram                  62 non-null     object 
dtypes: float64(1), int64(5), object(5)
memory usage: 5.5+ KB


In [11]:
data.head()

,Product Name,Product URL,Brand,Sale Price,Mrp,Discount Percentage,Number Of Ratings,Number Of Reviews,Upc,Star Rating,Ram
0,"APPLE iPhone 8 Plus (Gold, 64 GB)",https://www.flipkart.com/apple-iphone-8-plus-g...,Apple,49900,49900,0,3431,356,MOBEXRGV7EHHTGUH,4.6,2 GB
1,"APPLE iPhone 8 Plus (Space Grey, 256 GB)",https://www.flipkart.com/apple-iphone-8-plus-s...,Apple,84900,84900,0,3431,356,MOBEXRGVAC6TJT4F,4.6,2 GB
2,"APPLE iPhone 8 Plus (Silver, 256 GB)",https://www.flipkart.com/apple-iphone-8-plus-s...,Apple,84900,84900,0,3431,356,MOBEXRGVGETABXWZ,4.6,2 GB
3,"APPLE iPhone 8 (Silver, 256 GB)",https://www.flipkart.com/apple-iphone-8-silver...,Apple,77000,77000,0,11202,794,MOBEXRGVMZWUHCBA,4.5,2 GB
4,"APPLE iPhone 8 (Gold, 256 GB)",https://www.flipkart.com/apple-iphone-8-gold-2...,Apple,77000,77000,0,11202,794,MOBEXRGVPK7PFEJZ,4.5,2 GB


In [ ]:
data.isnull().sum()

,0
Product Name,0
Product URL,0
Brand,0
Sale Price,0
Mrp,0
Discount Percentage,0
Number Of Ratings,0
Number Of Reviews,0
Upc,0
Star Rating,0


the dataset does not have any missing values

In [12]:
data.describe()

,Sale Price,Mrp,Discount Percentage,Number Of Ratings,Number Of Reviews,Star Rating
count,62.000000,62.000000,62.000000,62.000000,62.000000,62.000000
mean,80073.887097,88058.064516,9.951613,22420.403226,1861.677419,4.575806
std,34310.446132,34728.825597,7.608079,33768.589550,2855.883830,0.059190
min,29999.000000,39900.000000,0.000000,542.000000,42.000000,4.500000
25%,49900.000000,54900.000000,6.000000,740.000000,64.000000,4.500000
50%,75900.000000,79900.000000,10.000000,2101.000000,180.000000,4.600000
75%,117100.000000,120950.000000,14.000000,43470.000000,3331.000000,4.600000
max,140900.000000,149900.000000,29.000000,95909.000000,8161.000000,4.700000


# **TOP HIGHEST RATED IPHONES ON FLIPKART IN INDIA**

In [13]:
highest_rated=data.sort_values(by = ["Star Rating"],ascending = False)

highest_rated = highest_rated.head(10)

In [14]:
print(highest_rated['Product Name'])

15                APPLE iPhone 11 Pro Max (Gold, 64 GB)
20      APPLE iPhone 11 Pro Max (Midnight Green, 64 GB)
17          APPLE iPhone 11 Pro Max (Space Grey, 64 GB)
16     APPLE iPhone 11 Pro Max (Midnight Green, 256 GB)
14               APPLE iPhone 11 Pro Max (Gold, 256 GB)
0                     APPLE iPhone 8 Plus (Gold, 64 GB)
12    Apple iPhone XR (Black, 128 GB) (Includes EarP...
11    Apple iPhone XR (Coral, 128 GB) (Includes EarP...
9     Apple iPhone XR ((PRODUCT)RED, 128 GB) (Includ...
1              APPLE iPhone 8 Plus (Space Grey, 256 GB)
Name: Product Name, dtype: object


In [15]:
print(highest_rated)

                                         Product Name  \
15              APPLE iPhone 11 Pro Max (Gold, 64 GB)   
20    APPLE iPhone 11 Pro Max (Midnight Green, 64 GB)   
17        APPLE iPhone 11 Pro Max (Space Grey, 64 GB)   
16   APPLE iPhone 11 Pro Max (Midnight Green, 256 GB)   
14             APPLE iPhone 11 Pro Max (Gold, 256 GB)   
0                   APPLE iPhone 8 Plus (Gold, 64 GB)   
12  Apple iPhone XR (Black, 128 GB) (Includes EarP...   
11  Apple iPhone XR (Coral, 128 GB) (Includes EarP...   
9   Apple iPhone XR ((PRODUCT)RED, 128 GB) (Includ...   
1            APPLE iPhone 8 Plus (Space Grey, 256 GB)   

                                          Product URL  Brand  Sale Price  \
15  https://www.flipkart.com/apple-iphone-11-pro-m...  Apple      117100   
20  https://www.flipkart.com/apple-iphone-11-pro-m...  Apple      117100   
17  https://www.flipkart.com/apple-iphone-11-pro-m...  Apple      117100   
16  https://www.flipkart.com/apple-iphone-11-pro-m...  Apple      13

# HOW MANY RATINGS DO THE HIGHEST RATED IPHONES ON FLIPKART HAVE?

In [81]:
# Step 1: Group by product name and sum the number of ratingss
iphone_Starrating = data.groupby("Product Name")["Star Rating"].sum().reset_index()

# Step 2: Sort in descending order (highest ratings first)
iphone_Starrating = iphone_Starrating.sort_values(by="Star Rating", ascending=False)

# Step 3: Select only the top 1
top_Starrating = iphone_Starrating.head(1)
print(top_Starrating)

                                        Product Name  Star Rating
13  APPLE iPhone 11 Pro Max (Midnight Green, 256 GB)          4.7


In [21]:
iphones = highest_rated['Product Name'].value_counts()
label = iphones.index
counts = highest_rated["Number Of Ratings"]
figure = px.bar(highest_rated, x=label, y=counts, title ="number of ratings of highest rated iphones")
figure.show()

# **top_number_of_ratings iphone on Flipkart**

# **WHICH IPHONE HAS THE HIGHEST NUMBER OF REVIEWS ON FLIPKART?**

In [19]:
data.head()

,Product Name,Product URL,Brand,Sale Price,Mrp,Discount Percentage,Number Of Ratings,Number Of Reviews,Upc,Star Rating,Ram
0,"APPLE iPhone 8 Plus (Gold, 64 GB)",https://www.flipkart.com/apple-iphone-8-plus-g...,Apple,49900,49900,0,3431,356,MOBEXRGV7EHHTGUH,4.6,2 GB
1,"APPLE iPhone 8 Plus (Space Grey, 256 GB)",https://www.flipkart.com/apple-iphone-8-plus-s...,Apple,84900,84900,0,3431,356,MOBEXRGVAC6TJT4F,4.6,2 GB
2,"APPLE iPhone 8 Plus (Silver, 256 GB)",https://www.flipkart.com/apple-iphone-8-plus-s...,Apple,84900,84900,0,3431,356,MOBEXRGVGETABXWZ,4.6,2 GB
3,"APPLE iPhone 8 (Silver, 256 GB)",https://www.flipkart.com/apple-iphone-8-silver...,Apple,77000,77000,0,11202,794,MOBEXRGVMZWUHCBA,4.5,2 GB
4,"APPLE iPhone 8 (Gold, 256 GB)",https://www.flipkart.com/apple-iphone-8-gold-2...,Apple,77000,77000,0,11202,794,MOBEXRGVPK7PFEJZ,4.5,2 GB


In [37]:
iphones = highest_rated['Product Name'].value_counts()
label = iphones.index
counts = highest_rated["Number Of Reviews"]
figure = px.bar(highest_rated, x=label, y=counts, title ="number of reviews of highest rated phones")
figure.show()

# **Iphone with highest number of reviews on FlipKart"**

In [38]:
iphones = data['Product Name'].value_counts()
label = iphones.index
counts = data["Number Of Reviews"]
figure = px.bar(iphones, x=label, y=counts, title ="Iphone with highest number of reviews on FlipKart")
figure.show()

In [77]:

# Step 1: Group by product name and sum the number of reviews
iphone_reviews = data.groupby("Product Name")["Number Of Reviews"].sum().reset_index()

# Step 2: Sort in descending order (highest reviews first)
iphone_reviews = iphone_reviews.sort_values(by="Number Of Reviews", ascending=False)

# Step 3: Select only the top 1
top_10_reviews = iphone_reviews.head(1)
print(top_10_reviews)

                       Product Name  Number Of Reviews
50  APPLE iPhone SE (Black, 128 GB)               8161


# **what is the relationship between sales price of iphone and number of ratings on flipkart**

In [39]:
data.head()

,Product Name,Product URL,Brand,Sale Price,Mrp,Discount Percentage,Number Of Ratings,Number Of Reviews,Upc,Star Rating,Ram
0,"APPLE iPhone 8 Plus (Gold, 64 GB)",https://www.flipkart.com/apple-iphone-8-plus-g...,Apple,49900,49900,0,3431,356,MOBEXRGV7EHHTGUH,4.6,2 GB
1,"APPLE iPhone 8 Plus (Space Grey, 256 GB)",https://www.flipkart.com/apple-iphone-8-plus-s...,Apple,84900,84900,0,3431,356,MOBEXRGVAC6TJT4F,4.6,2 GB
2,"APPLE iPhone 8 Plus (Silver, 256 GB)",https://www.flipkart.com/apple-iphone-8-plus-s...,Apple,84900,84900,0,3431,356,MOBEXRGVGETABXWZ,4.6,2 GB
3,"APPLE iPhone 8 (Silver, 256 GB)",https://www.flipkart.com/apple-iphone-8-silver...,Apple,77000,77000,0,11202,794,MOBEXRGVMZWUHCBA,4.5,2 GB
4,"APPLE iPhone 8 (Gold, 256 GB)",https://www.flipkart.com/apple-iphone-8-gold-2...,Apple,77000,77000,0,11202,794,MOBEXRGVPK7PFEJZ,4.5,2 GB


In [42]:
figure = px.scatter(data_frame = data, x='Number Of Ratings', y='Sale Price', size='Discount Percentage', trendline= 'ols', title='relationship between sales price and number of ratings')
figure.show()

There is a negative realtionship between the sale price of iphones and the number of ratings. Lower cost iphones have higher ratings.

# **what is the relationship between discount price of iphone and number of ratings on flipkart**

In [52]:
fig=px.scatter(data_frame=data, x= 'Number Of Ratings', y='Discount Percentage', size="Sale Price", trendline="ols", title="relationship between discount price and number of ratings")
fig.show()

There is a positive relationship with discount percentage and number of ratings. Higher the disocunt, higher the number of ratings.



# **Can you figure out the least expensive and most expensive iphones in the Indian market, along with all specifications?**


In [53]:
data.head()

,Product Name,Product URL,Brand,Sale Price,Mrp,Discount Percentage,Number Of Ratings,Number Of Reviews,Upc,Star Rating,Ram
0,"APPLE iPhone 8 Plus (Gold, 64 GB)",https://www.flipkart.com/apple-iphone-8-plus-g...,Apple,49900,49900,0,3431,356,MOBEXRGV7EHHTGUH,4.6,2 GB
1,"APPLE iPhone 8 Plus (Space Grey, 256 GB)",https://www.flipkart.com/apple-iphone-8-plus-s...,Apple,84900,84900,0,3431,356,MOBEXRGVAC6TJT4F,4.6,2 GB
2,"APPLE iPhone 8 Plus (Silver, 256 GB)",https://www.flipkart.com/apple-iphone-8-plus-s...,Apple,84900,84900,0,3431,356,MOBEXRGVGETABXWZ,4.6,2 GB
3,"APPLE iPhone 8 (Silver, 256 GB)",https://www.flipkart.com/apple-iphone-8-silver...,Apple,77000,77000,0,11202,794,MOBEXRGVMZWUHCBA,4.5,2 GB
4,"APPLE iPhone 8 (Gold, 256 GB)",https://www.flipkart.com/apple-iphone-8-gold-2...,Apple,77000,77000,0,11202,794,MOBEXRGVPK7PFEJZ,4.5,2 GB


In [55]:
most_expensive = data.sort_values(by = ["Sale Price"], ascending = False)
most_expensive.head(1)

,Product Name,Product URL,Brand,Sale Price,Mrp,Discount Percentage,Number Of Ratings,Number Of Reviews,Upc,Star Rating,Ram
24,"APPLE iPhone 12 Pro (Silver, 512 GB)",https://www.flipkart.com/apple-iphone-12-pro-s...,Apple,140900,149900,6,542,42,MOBFWBYZ5UY6ZBVA,4.5,4 GB


In [58]:
least_expensive = data.sort_values(by = ["Sale Price"], ascending=True)
least_expensive.head(1)

,Product Name,Product URL,Brand,Sale Price,Mrp,Discount Percentage,Number Of Ratings,Number Of Reviews,Upc,Star Rating,Ram
52,"APPLE iPhone SE (White, 64 GB)",https://www.flipkart.com/apple-iphone-se-white...,Apple,29999,39900,24,95807,8154,MOBFWQ6BGWDVGF3E,4.5,2 GB


In [63]:
#alternately run the below code for direct pulling of data
#most_expensive = data.loc[data["Sale Price"].idxmax()]
#least_expensive= data.loc[data["Sale Price"].idxmin()]


most_expensive = data.loc[data["Sale Price"].idxmax()]
print("Most Expensive Product:")
print(most_expensive)




Most Expensive Product:
Product Name                        APPLE iPhone 12 Pro (Silver, 512 GB)
Product URL            https://www.flipkart.com/apple-iphone-12-pro-s...
Brand                                                              Apple
Sale Price                                                        140900
Mrp                                                               149900
Discount Percentage                                                    6
Number Of Ratings                                                    542
Number Of Reviews                                                     42
Upc                                                     MOBFWBYZ5UY6ZBVA
Star Rating                                                          4.5
Ram                                                                 4 GB
Name: 24, dtype: object


In [61]:
least_expensive= data.loc[data["Sale Price"].idxmin()]
print("Least Expensive Product:")
print(least_expensive)

Least Expensive Product:
Product Name                              APPLE iPhone SE (White, 64 GB)
Product URL            https://www.flipkart.com/apple-iphone-se-white...
Brand                                                              Apple
Sale Price                                                         29999
Mrp                                                                39900
Discount Percentage                                                   24
Number Of Ratings                                                  95807
Number Of Reviews                                                   8154
Upc                                                     MOBFWQ6BGWDVGF3E
Star Rating                                                          4.5
Ram                                                                 2 GB
Name: 52, dtype: object


# **What are the top 10 highest rated iphone on the ecomm site Flipkart? **
APPLE iPhone 11 Pro Max (Gold, 64 GB)
APPLE iPhone 11 Pro Max (Midnight Green, 64 GB)
APPLE iPhone 11 Pro Max (Space Grey, 64 GB)
APPLE iPhone 11 Pro Max (Midnight Green, 256 GB)
APPLE iPhone 11 Pro Max (Gold, 256 GB)
APPLE iPhone 8 Plus (Gold, 64 GB)
Apple iPhone XR (Black, 128 GB) (Includes EarP...
Apple iPhone XR (Coral, 128 GB) (Includes EarP...
Apple iPhone XR ((PRODUCT)RED, 128 GB) (Includ...
APPLE iPhone 8 Plus (Space Grey, 256 GB)
    





# **how many ratings do the highest rated iphones on Flipkart have?  **
APPLE iPhone 11 Pro Max (Midnight Green, 256 GB)          4.7




# **Which iphone has the highest number of reviews on FlipKart? **  
 APPLE iPhone SE (Black, 128 GB) has highest reviews of 8161



# **What is the relationship between the sales price of iphones and number of ratings on FlipKart? **  
There is a negative relationship between the sale price of iphones and the number of ratings. Lower cost iphones have higher ratings.


# **What is the relationship between discount percentage and the number of ratings of iphones on FlipKart? **  
here is a positive relationship with discount percentage and number of ratings. Higher the disocunt, higher the number of ratings.


# **Can you figure out the least expensive and the most expensive iPhones in the indian market alongwith all their specifications?**

**Least Expensive Product:**
Product Name                              APPLE iPhone SE (White, 64 GB)
Product URL            https://www.flipkart.com/apple-iphone-se-white...
Brand                                                              Apple
Sale Price                                                         29999
Mrp                                                                39900
Discount Percentage                                                   24
Number Of Ratings                                                  95807
Number Of Reviews                                                   8154
Upc                                                     MOBFWQ6BGWDVGF3E
Star Rating                                                          4.5
Ram                                                                 2 GB


**Most Expensive Product:**
Product Name                        APPLE iPhone 12 Pro (Silver, 512 GB)
Product URL            https://www.flipkart.com/apple-iphone-12-pro-s...
Brand                                                              Apple
Sale Price                                                        140900
Mrp                                                               149900
Discount Percentage                                                    6
Number Of Ratings                                                    542
Number Of Reviews                                                     42
Upc                                                     MOBFWBYZ5UY6ZBVA
Star Rating                                                          4.5
Ram                                                                 4 GB




